homeassistant/components/weather/openweathermap.py

```diff
@@ -149,6 +149,14 @@ def attribution(self):
     def forecast(self):
         """Return the forecast array."""
         data = []
+
+        def calc_precipitation(rain, snow):
+            rain_value = 0 if rain is None else rain
+            snow_value = 0 if snow is None else snow
+            if round(rain_value + snow_value, 1) == 0:
+                return None
+            return round(rain_value + snow_value, 1)
```

Comment:
Should the precipitation be `None` if both `rain` and `snow` are 0 ?

Reply:
I think yes. Otherwise, we need to change the weather card to not display zeros.

Reply:
Should we not show 0 if the forecast is 0 snow and 0 rain?

Reply:
It should be `None`.